packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,319 @@
+export const count = jobs.createJob<number, number>(
+  // Receive a context that contains additional methods, like progress, logger and channels.
+  (input: number, { progress, channels }) => new Observable<number>(o => {
+    let i = 0;
+    function doCount() {
+      o.next(i++);
+      progress(i / input);
+      channels['side'].next('more');
+  
+      if (i < input) {
+        setTimeout(doCount, 100);
+      } else {
+        o.complete();
+      }
+    }
+    setTimeout(doCount, 100);
+  }),
+  {
+      input: { type: 'number' },
+      output: { type: 'number' },
+      channels: {
+        'side': { type: 'string', const: 'more' },
+      },
+  },
+);
+
+// Get a hold of a scheduler that refers to the job above.
+declare const scheduler: jobs.Scheduler;
+
+const job = scheduler.schedule('count');
+job.channels['side'] && job.channels['side'].subscribe(x => console.log(x));
+const c = job.getChannel<string>('side', { type: 'string' });
+if (c) {
+  c.subscribe((x: string) => console.log(x));
+}
+```
+
+## Job Extension
+A Job can extend other jobs, meaning their inputs and outputs have to be a superset of the job 
+extended. This is useful for making compatible jobs that can be replaced by others.
+
+## Job Groups
+Groups are an additional way to synchronize and execute logic on scheduling the job, from the 
+side of the scheduler. This logic will always apply in the thread the job is scheduled, and not
+where the job is created/registered.
+
+Job groups are a job helper that redirect to different jobs. To create a job group, use the
+`createGroup()` function:
+
+```typescript
+import { jobs } from '@angular-devkit/core';
+
+// A group that installs node modules.
+const group = jobs.createGroup({
+  name: 'node-install',
+  input: {
+    properties: {
+      moduleName: { type: 'string' },
+    },
+  },
+  output: { type: 'boolean' },
+});
+
+// Extending makes sure the input and output matches the group.
+const npmInstall = jobs.createJob(/* ... */, { name: 'npm-install', extends: group });
+const yarnInstall = jobs.createJob(/* ... */, { name: 'yarn-install', extends: group });
+const pnpmInstall = jobs.createJob(/* ... */, { name: 'pnpm-install', extends: group });
+
+declare const registry: jobs.SimpleJobRegistry;
+registry.register(group);
+registry.register(npmInstall);
+registry.register(yarnInstall);
+registry.register(pnpmInstall);
+
+// Default to npm.
+group.setDefaultDelegate(npmInstall.name);
+// If the user is asking for yarn over npm, uses it.
+group.addConditionalDelegate(() => userWantsYarn, yarnInstall.name);
+```
+
+## Execution Strategy
+
+Jobs are always run in parallel and will always start, but many helper functions are provided 
+when creating a job to help you control the execution strategy;
+
+1. `serialize()`. Multiple runs of this job will be queued with each others.
+1. `reuse(replayEvents = false)`. Jobs with this strategy will reuse an already running job. If 
```

Comment:
I don't understand the reuse strategy. A job has a single input. How can a running job receive more inputs?

Reply:
It was removed. It doesn't make sense.

Reply:
Essentially it was a "Never rerun the job, just reuse the first run".